static_src/stores/login_store.js

```diff
@@ -7,25 +7,44 @@ import BaseStore from './base_store.js';
 import { loginActionTypes } from '../constants.js';
 
 
-class LoginStore extends BaseStore {
+export class LoginStore extends BaseStore {
   constructor() {
     super();
     this.subscribe(() => this._registerToActions.bind(this));
+    // TODO this should probably be false, but we need to account for the
+    // initial state (not loaded/unknown).
     this._isAuthenticated = true;
+    this._error = null;
   }
 
   _registerToActions(action) {
     switch (action.type) {
+      case loginActionTypes.FETCH_STATUS:
+        // Reset any error
+        this._error = null;
+        this.emitChange();
+        break;
+
       case loginActionTypes.RECEIVED_STATUS:
-        this._isAuthenticated = action.status;
+        this._isAuthenticated = action.authStatus.status === 'authorized';
         this.emitChange();
         break;
 
+      case loginActionTypes.ERROR_STATUS:
+        // Login status is unknown. If we have a login status, leave it as is
+        // and hope things go smooth. A notification is handled elsewhere.
```

Comment:
Yeah, it happens in `checkAuth`, but I might update the comment to be more like:
> If necessary, the action caller is responsible for notifying the user of an error.

That way the comment doesn't need to know where elsewhere is.